src/rgw/rgw_data_sync.cc

```diff
@@ -1877,15 +1932,43 @@ class RGWDataSyncCR : public RGWCoroutine {
         return set_cr_error(retcode);
       }
 
+      if ((rgw_data_sync_info::SyncState)sync_status.sync_info.state !=
+	  rgw_data_sync_info::StateSync) {
+	init_lease.reset(
+	  RGWInitDataSyncStatusCoroutine::continuous_lease_cr(sc, this));
+
+	yield spawn(init_lease.get(), false);
+	while (!init_lease->is_locked()) {
+	  if (init_lease->is_done()) {
+	    tn->log(5, "ERROR: failed to take data sync status lease");
+	    set_status("lease lock failed, early abort");
+	    drain_all();
+	    return set_cr_error(init_lease->get_ret_status());
+	  }
+	  tn->log(5, "waiting on data sync status lease");
+	  yield set_sleeping(true);
+	}
+	tn->log(5, "acquired data sync status lease");
+
+	// Reread sync status now that we've acquired the lock!
+	yield call(new RGWReadDataSyncStatusCoroutine(sc, &sync_status, &obj_version));
```

Comment:
at this point, `obj_version` contains the `read_version` from the earlier call to `RGWReadDataSyncStatusCoroutine`. passing this to another read will add a `cls_version_check()` that fails with `ECANCELED` if there was a racing write

we don't want that here, we just want to re-read the latest version - so we either need to call `obj_version.clear()` before this one, or pass `objv=nullptr` for the first call

Reply:
Done.